src/Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -477,5 +491,33 @@ void ISupportNuget.RegisterResolvedPackageReferences(IReadOnlyList<ResolvedPacka
 
         public IEnumerable<string> RestoreSources =>
             PackageRestoreContext.RestoreSources;
+
+        public Task HandleAsync(RequestValueNames command, KernelInvocationContext context)
+        {
+            context.Publish(new ValueNamesProduced(GetVariableNames(), command));
+            return Task.CompletedTask;
+        }
+
+        public Task HandleAsync(RequestValue command, KernelInvocationContext context)
+        {
+            if (TryGetVariable(command.Name, out var value))
+            {
+                var formattedValues = new List<FormattedValue>();
+                if (command.MimeTypes?.Any() == true)
+                {
+                    formattedValues.AddRange(command.MimeTypes.Select(mimeType => new FormattedValue(mimeType, value?.ToDisplayString(mimeType))));
```

Comment:
What if a given MIME type isn't supported?

Reply:
probably we should throw

Reply:
we throw now